core/src/window_service.rs

```diff
@@ -228,13 +228,6 @@ where
                                     }
                                 };
                                 if shred_filter(&shred, last_root) {
-                                    // Mark slot as dead if the current shred is on the boundary
```

Comment:
I opted to just drop this check since it doesn't provide much value. Think it deserves a feature switch?